packages/git/src/node/dugite-git.ts

```diff
@@ -226,6 +226,33 @@ export class DugiteGit implements Git {
         });
     }
 
+    async diff(repository: Repository, options?: Git.Options.Diff): Promise<GitFileChange[]> {
+        const args = [
+            'diff',
+            '--name-status',
+            '-B', // This makes sure, that rewrites (such as rename) will be a pair of delete and create instead of the `R36 oldFileName newFileName` one-liner format.
```

Comment:
but don't we lose the information that it was a move, then?

Reply:
Yes, but I would expect, that it is how GitHub behaves. I might be wrong though.

Just to clarify my understanding on this; if you have a file (`myFile.txt`) with content `book` and you rename it to `myFile2.txt` but do not alter the content, then it is a rename, and you will see as a rename.

But, if you have a file (`myFile.txt`) with content `book` and you rename the file to `myFile2.txt`. **And** you change the content to `pool` then with the `-B` flag it will be a `deleted` - `myFile.txt` and a `new` - `myFile2.txt` change.

I need to add more tests...

Reply:
It's just that git has some smartness in detecting moves and I think we should use that.

Reply:
One could check [this example](https://github.com/theia-ide/theia/pull/1092#discussion_r163192685) from the command line with:

```
git clone https://github.com/eclipse/eclipse.jdt.ls.git \
&& git checkout -b docker \
&& git checkout -b Java9 \
&& git diff --name-status docker..Java9
```